# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/patches/samba-strlcat_strlcpy.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- samba-3.2.4/source/client/mount.cifs.c.orig	2008-09-18 08:49:02.000000000 +0200
+++ samba-3.2.4/source/client/mount.cifs.c	2008-11-02 14:02:03.000000000 +0100
@@ -89,6 +89,10 @@
 char * domain_name = NULL;
 char * prefixpath = NULL;
 
+#ifdef __UCLIBC__
+/* strlcpy/strlcat defined in <string.h> */
+#else
+
 /* glibc doesn't have strlcpy, strlcat. Ensure we do. JRA. We
  * don't link to libreplace so need them here. */
 
@@ -126,6 +130,7 @@
 	}
 	return ret;
 }
+#endif
 
 /* BB finish BB
 
